pos_main.py

```diff
@@ -0,0 +1,162 @@
+"""This script facilitates plus one scaling for disperse volume"""
+
+from scale.gf_plus_one_scale import *
+from test_data.gf_plus_one_test_data import *
+from scale.gf_logs import *
+from scale.gf_exceptions import *
+from scale.gf_api import *
+import os
+
+
+def gf_get_volume_and_bricks():
+    """Get volume name, new node and file name from user.
+    """
+    volname = input("Enter name of the volume: ")
+    filename = input("Enter file name which contains new bricks: ")
+    hostname = input("Enter IP/hostname of new node: ")
+
+    if not volname or not filename or not hostname:
+        glogger.debug('Invalid volume name and filename')
+        raise ValueError("volname or filename or hostname is not correct")
+
+    return volname, filename, hostname
+
+
+def gluster_volume_commit_all(volname, bricks_migration_map):
+    """Commit physical replacment of all the bricks.
+
+    Input:
+    volname: Name of the volume
+    bricks_migration_map: Physical migration map of old and new bricks.
+    """
+    for old_brick, new_brick in bricks_migration_map:
+        glogger.info(f'Commiting replace bricks for old_brick = {old_brick} and new_brick = {new_brick}')
+        print(f'Commiting replace bricks for old_brick = {old_brick} and new_brick = {new_brick}')
+        try:
+            gluster_volume_commit_drive_swap(volname, old_brick, new_brick)
+        except GfCommandFailed as er:
+            glogger.exception(f'replace brick failed for {old_brick} and {new_brick}')
+            continue
+        else:
+            glogger.info(f'replace brick is done for {old_brick} and {new_brick}')
+
+
+def gf_swap_all_bricks(volname, bricks_migration_map):
+    """Swap all the bricks.
+
+    This function helps to swap all the bricks first and then commit
+    the physical migration in one shot.
+    Input:
+    volname: Name of the volume
+    bricks_migration_map: Physical migration map of old and new bricks.
+    """
+    while True:
+        print("Swap following disks : ")
+        for br in bricks_migration_map:
+            print(br)
+        done = input("Have you finished swapping all the disks: (y/n) ")
+        if done == 'y' or done == "Y":
+            gluster_volume_commit_all(volname, bricks_migration_map)
+        else:
+            continue
+
+
+def gf_swap_one_by_one_brick(volname, bricks_migration_map):
+    """Swap bricks one by one.
+
+    Ask user to swap a pair of old and new brick and then
+    commit that migration
+    Input:
+    volname: Name of the volume
+    bricks_migration_map: Physical migration map of old and new bricks.
+    """
+    for old_brick, new_brick in bricks_migration_map:
+        while True:
+            try:
+                print(f'Swap {old_brick} and {new_brick}')
+                done = input("Have you finished swapping above disks: (y/n) ")
+                if done == 'y' or done == "Y":
```

Comment:
This pattern is repeated in multiple places, convert this to a function

```
def yes(value):
    return value.lower() in ["y", "yes"]
```

Reply:
Done